bundles/s3-bundle/src/main/java/org/apache/gravitino/fileset/s3/S3FileSystemProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.fileset.s3;
+
+import java.io.IOException;
+import java.net.URI;
+import org.apache.gravitino.catalog.hadoop.FileSystemProvider;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+public class S3FileSystemProvider implements FileSystemProvider {
+
+  @Override
+  public FileSystem getFileSystem(Configuration configuration, Path path) throws IOException {
+    URI uri = path.toUri();
+    if (uri.getScheme() == null || !uri.getScheme().equals("s3a")) {
+      throw new IllegalArgumentException("The path should be a S3 path.");
+    }
+
+    // TODO Check whether S3 related configurations are set such as filesystem.s3a.impl, access key,
+    //  secret key, etc.
+    return FileSystem.get(uri, configuration);
+  }
+
+  @Override
+  public String getScheme() {
+    return "s3a";
+  }
+}
```

Comment:
Can you please remove gcs, s3 related code, we can add them later on.

Reply:
I see.